src/components/program/ProgramPage.jsx

```diff
@@ -49,12 +49,15 @@ class ProgramPage extends Component {
       programDiscussions,
       liveSettings,
     } = this.props;
+
     if (enrolledPrograms && enrolledPrograms !== prevProps.enrolledPrograms) {
       this.validateProgramAccess(enrolledPrograms);
     }
-    if (tabViewEnabled && tabViewEnabled !== prevProps.tabViewEnabled) {
-      // eslint-disable-next-line max-len
-      const switchTabView = tabViewEnabled && (programDiscussions.configured || liveSettings.configured);
+    if ((tabViewEnabled && tabViewEnabled !== prevProps.tabViewEnabled)
+      || programDiscussions.configured !== prevProps.programDiscussions.configured
+      || liveSettings.configured !== prevProps.liveSettings.configured) {
+      const switchTabView = tabViewEnabled
+        && (programDiscussions.configured || liveSettings.configured);
```

Comment:
Didn't the API return the data that tab view is enabled for this program? why you are checking the logic here?

Reply:
there are multiple apis each returning its own part of data what is configured and whats not. frontend needs to check commulatively if we need to switch to tab view or not so lets suppose

```
flag : enabled
live: disabled
discussion: disbaled
```

so although flag is enabled but we cant switch to tab view.